### PR TITLE
Upgrade hashing to rs256 -- needs synchronous change on frontend

### PR DESCRIPTION
**Pull request by davidascher**
Also make middleware functions less error prone -- make explicit whether they require auth or not.

### CODE REVIEW

File: test/cases.js [hunk 1]
```diff
@@ -7,6 +7,22 @@ let should = chai.should();
 chai.use(chaiHttp);
 chai.use(chaiHelpers);
 
+// let jwt = require("jsonwebtoken");
```
Comment (dethe) [inline test/cases.js]: Shall we delete this commented-out code?

File: test/cases.js [hunk 2]
```diff
@@ -205,11 +221,16 @@ describe("Cases", () => {
   });
   describe("Get case with authentication", () => {
     it("should not fail when logged in", async () => {
-      const res = await chai
-        .getJSON("/case/100")
-        .set("Authorization", "Bearer " + tokens.user_token);
-      res.body.OK.should.equal(true);
-      res.should.have.status(200);
+      console.log("TOKEN", tokens.user_token);
```
Comment (dethe) [inline test/cases.js]: Stray console.log?

Comment (davidascher) [inline test/cases.js]: duh.

File: test/cases.js (hunk 1 above)
Comment (davidascher) [inline test/cases.js]: duh.

**Pull request merged**